pandas/tests/scalar/timestamp/test_timestamp.py

```diff
@@ -780,6 +780,13 @@ def test_hash_equivalent(self):
         stamp = Timestamp(datetime(2011, 1, 1))
         assert d[stamp] == 5
 
+    def test_tz_convert_freq(self, tz_naive_fixture):
+        # GH25241
+        t1 = Timestamp('2019-01-01 10:00', freq='H')
+        assert t1.tz_localize(tz=tz_naive_fixture).freq == t1.freq
+        t2 = Timestamp('2019-01-02 12:00', tz='UTC', freq='T')
+        assert t2.tz_convert(tz='UTC') == t2
```

Comment:
I think you need to assert that the `freq` attributes are equal, as opposed to comparing the timestamps.

Timestamp `freq` values do not influence [equality comparisons](https://github.com/pandas-dev/pandas/blob/2448e5229683acbe7de57f2d53065247aa085b1f/pandas/_libs/tslibs/timestamps.pyx#L216) or hash values.  Here's a quick example that shows this:

```
>>> t1 = pandas.Timestamp('2018-01-01 10:00', freq='H') 
>>> t2 = pandas.Timestamp('2018-01-01 10:00', freq='T') 
>>> t1 == t2
True
>>> t1.freq == t2.freq
False
```

Reply:
ahh, nice catch!! 
```python
t1 = pd.Timestamp('2018-01-01 10:00', freq='H')
t2 = pd.Timestamp('2018-01-01 10:00')])
```
and `t1 != t2`! thanks! @nmusolino